pkg/publicshare/manager/owncloudsql/owncloudsql.go

```diff
@@ -0,0 +1,544 @@
+		}
+	}
+	if resourceFilters != "" {
+		query = fmt.Sprintf("%s AND (%s)", query, resourceFilters)
+		params = append(params, resourceParams...)
+	}
+	if ownerFilters != "" {
+		query = fmt.Sprintf("%s AND (%s)", query, ownerFilters)
+		params = append(params, ownerParams...)
+	}
+	if creatorFilters != "" {
+		query = fmt.Sprintf("%s AND (%s)", query, creatorFilters)
+		params = append(params, creatorParams...)
+	}
+	if storageFilters != "" {
+		query = fmt.Sprintf("%s AND (%s)", query, storageFilters)
+		params = append(params, storageParams...)
+	}
+
+	rows, err := m.db.Query(query, params...)
+	if err != nil {
+		return nil, err
+	}
+	defer rows.Close()
+
+	var s DBShare
+	shares := []*link.PublicShare{}
+	for rows.Next() {
+		if err := rows.Scan(&s.UIDOwner, &s.UIDInitiator, &s.ShareWith, &s.FileSource, &s.ItemType, &s.Token, &s.Expiration, &s.ShareName, &s.ID, &s.STime, &s.Permissions, &s.ItemStorage); err != nil {
+			continue
+		}
+		var cs3Share *link.PublicShare
+		if cs3Share, err = m.ConvertToCS3PublicShare(ctx, s); err != nil {
+			return nil, err
+		}
+		if publicshare.IsExpired(*cs3Share) {
+			_ = m.cleanupExpiredShares()
+		} else {
+			if cs3Share.PasswordProtected && sign {
+				if err := publicshare.AddSignature(cs3Share, strings.TrimPrefix(s.ShareWith, "1|")); err != nil {
+					return nil, err
+				}
+			}
+			shares = append(shares, cs3Share)
+		}
+	}
+	if err = rows.Err(); err != nil {
+		return nil, err
+	}
+
+	return shares, nil
+}
+
+func (m *mgr) RevokePublicShare(ctx context.Context, u *user.User, ref *link.PublicShareReference) error {
+	uid := u.Username
+	query := "delete from oc_share where "
+	params := []interface{}{}
+
+	switch {
+	case ref.GetId() != nil && ref.GetId().OpaqueId != "":
+		query += "id=? AND (uid_owner=? or uid_initiator=?)"
+		params = append(params, ref.GetId().OpaqueId, uid, uid)
+	case ref.GetToken() != "":
+		query += "token=? AND (uid_owner=? or uid_initiator=?)"
+		params = append(params, ref.GetToken(), uid, uid)
+	default:
+		return errtypes.NotFound(ref.String())
+	}
+
+	stmt, err := m.db.Prepare(query)
+	if err != nil {
+		return err
+	}
+	res, err := stmt.Exec(params...)
+	if err != nil {
+		return err
+	}
+
+	rowCnt, err := res.RowsAffected()
+	if err != nil {
+		return err
+	}
+	if rowCnt == 0 {
+		return errtypes.NotFound(ref.String())
+	}
+	return nil
+}
+
+func (m *mgr) GetPublicShareByToken(ctx context.Context, token string, auth *link.PublicShareAuthentication, sign bool) (*link.PublicShare, error) {
+
+	ps, err := m.getByToken(ctx, token)
+	if err != nil {
+		return nil, err
+	}
+
+	if publicshare.IsExpired(ps.PublicShare) {
+		if err := m.cleanupExpiredShares(); err != nil {
+			return nil, err
+		}
+		return nil, errtypes.NotFound("public share has expired")
+	}
+
+	if ps.PublicShare.PasswordProtected {
+		if !publicshare.Authenticate(&ps.PublicShare, ps.Password, auth) {
+			return nil, errtypes.InvalidCredentials("access denied")
+		}
+		/*
+			if sign {
```

Comment:
Hm, is there a reason why we don't add the signature when getting the share by token?

Reply:
dunno, the cs3 driver does not do it as well.